node/task.ts

```diff
@@ -725,36 +728,105 @@ export function resolve(...pathSegments: any[]): string {
+            if (recursive) {
+                command += ' /s';
+            }
+            if (allFiles) {
+                command += ' /a';
+            }
+        }
+        else {
+            command = 'ls ' + path;
+            if (recursive) {
+                command += ' -R';
+            }
+            if (allFiles) {
+                command += ' -A';
+            }
+        }
+        const childProcessResponse: string = childProcess.execSync(command).toString();
+        childProcessResponse.split(os.EOL).forEach((file: string) => {
+            if (filePaths.indexOf(file) < 0) {
+                filePaths.push(file);
+            }
+        });
+    });
+
+    return filePaths;
 }
 
 /**
  * Copies a file or folder.
  * 
  * @param     source     source path
  * @param     dest       destination path
- * @param     options    string -r, -f or -rf for recursive and force 
+ * @param     recursive  Copy recursively
+ * @param     force      Force copy
  * @param     continueOnError optional. whether to continue on error
  */
-export function cp(source: string, dest: string, options?: string, continueOnError?: boolean): void {
-    if (options) {
-        shell.cp(options, source, dest);
+export function cp(source: string, dest: string, recursive: boolean = false, force: boolean = false, continueOnError?: boolean): void {
+    try {
+        if (!fs.existsSync(source)) {
+            throw new Error(loc('LIB_OperationFailed', source + ' doesnt exist'));
+        }
+        if (getPlatform() == Platform.Windows) {
+            if (fs.statSync(source).isDirectory()) {
+                mkdirP(dest);
+                let command: string = 'robocopy ' + source + ' ' + dest;
+                
+                if (recursive) {
+                    command += ' /e';
+                }
+                if (force) {
+                    command += ' /is /it';
+                }
+
+                childProcess.execSync(command);
+            }
+            else {
+                // Copy individual file over
+                if (fs.existsSync(dest)) {
+                    if (force) {
+                        fs.unlinkSync(dest);
+                    }
+                    else {
+                        // If file exists and we're not overwriting, just return. 
+                        return;
+                    }
+                }
+                const command: string = 'echo F | xcopy ' + source + ' ' + dest;
+                childProcess.execSync(command);
+            }
+        }
+        else {
+            let command: string = 'cp';
+            if (recursive) {
+                command += ' -R';
+            }
+            if (force) {
+                command += ' -f';
+            }
+            command += ' ' + source + ' ' + dest;
+            childProcess.execSync(command);
+        }
     }
-    else {
-        shell.cp(source, dest);
+    catch (err) {
+        if (!continueOnError) {
```

Comment:
Should we still write to the log that there was an error?

Reply:
Yeah (here and other places). Good call, I'll add that